### PR TITLE
Convert @bugsnag/plugin-network-breadcrumbs to TypeScript

### PR DESCRIPTION
**Pull request by AnastasiiaSvietlova**
## Goal

Convert package to TypeScript and ES Modules

## Changeset

Refactor @bugsnag/plugin-network-breadcrumbs

### CODE REVIEW

**Comment by gingerbenw** [comment]
@yousif-bugsnag I've made quite a few changes from @AnastasiiaSvietlova's original PR, so would appreciate another pair of eyes 🙏🏻

File: packages/react-native/window.d.ts
Comment (yousif-bugsnag) [inline packages/react-native/window.d.ts:1]: I don't think I understand why this is necessary?

Comment (gingerbenw) [reply packages/react-native/window.d.ts:1]: it would otherwise fail to compile because window is an unknown type in react native

Comment (yousif-bugsnag) [reply packages/react-native/window.d.ts:1]: oh, I see - I think we could fix that by declaring our own interface in plugin-network-breadcrumbs, e.g:

```ts
interface GlobalWithFetchAndXHR {
  fetch: typeof fetch
  XMLHttpRequest: typeof XMLHttpRequest
}

export default (_ignoredUrls = [], win: GlobalWithFetchAndXHR = window): Plugin => {
  // ...
}
```

**Review by yousif-bugsnag — APPROVED** [review]

**Pull request merged**